app/models/auth_sources/auth_source_ldap.rb

```
@@ -144,33 +177,51 @@ def store_avatar avatar
-    object_filter = Net::LDAP::Filter.eq("objectClass", "*")
+  def search_for_user_entry(login)
+    ldap_fluff = LdapFluff.new(self.to_config)
+    return nil unless ldap_fluff.valid_user?(login)
+
+    if attr_login.blank?
+      login_filter = ldap_fluff.ldap.member_service.name_filter(login)
+    else
+      login_filter = Net::LDAP::Filter.eq(attr_login, login)
+    end
+
+    object_filter = Net::LDAP::Filter.eq('objectClass', '*')
     object_filter = object_filter & Net::LDAP::Filter.construct(ldap_filter) unless ldap_filter.blank?
 
-    # search for a match for our authenticating user.
-    entries       = ldap_con.search(:base       => base_dn,
-                                    :filter     => object_filter & login_filter,
-                                    # only ask for the DN if on-the-fly registration is disabled
-                                    :attributes => required_ldap_attributes.values + optional_ldap_attributes.values)
-    unless ldap_con.get_operation_result.code == 0
-      logger.warn "Search Result: #{ldap_con.get_operation_result.code}"
-      logger.warn "Search Message: #{ldap_con.get_operation_result.message}"
+    entries = ldap_con(ldap_fluff).search(:base => base_dn, :filter => object_filter & login_filter)
+
+    unless ldap_con(ldap_fluff).get_operation_result.code == 0
+      logger.warn "Search Result: #{ldap_con(ldap_fluff).get_operation_result.code}"
+      logger.warn "Search Message: #{ldap_con(ldap_fluff).get_operation_result.message}"
     end
 
-    # we really care about one match, using the last one, hoping there is only one match :)
     entries ? entries.last : nil
   end
 
+  def update_usergroups(entry)
+    if entry.respond_to? :memberof
+      group_list = entry.memberof
+    elsif entry.respond_to? :ismemberof
+      group_list = entry.ismemberof
+    else
+      return
+    end
+
+    group_list.each do |name|
+      begin
+      Usergroup.find_by_name(name).refresh_ldap
+      rescue
```

should we at least log something to warning/debug if this fails?  Should we actually raise the error instead of catching it?